qadence/noise/readout.py

```diff
@@ -0,0 +1,209 @@
+from __future__ import annotations
+
+from collections import Counter
+from enum import Enum
+
+import numpy as np
+import torch
+from torch import Tensor
+from torch.distributions import normal, poisson, uniform
+
+from qadence.logger import get_logger
+
+logger = get_logger(__name__)
+
+
+class WhiteNoise(Enum):
+    """White noise distributions."""
+
+    UNIFORM = staticmethod(uniform.Uniform(low=0.0, high=1.0))
+    """Uniform white noise."""
+
+    GAUSSIAN = staticmethod(normal.Normal(loc=0.0, scale=1.0))
+    """Gaussian white noise."""
+
+    POISSON = staticmethod(poisson.Poisson(rate=0.1))
+    """Poisson white noise."""
+
+
+def bitstring_to_tensor(bitstring: str, output_type: str = "torch") -> np.array | Tensor:
+    """
+    A helper function to convert bit strings to torch.Tensor or numpy.array.
+
+    Args:
+        bitstring:  A str format of a bit string.
+        output_type: A str torch | numpy for the type of the output.
+        Default torch.
+
+    Returns:
+        A torch.Tensor or np.array out of the input bit string.
+    """
+    return (
+        torch.as_tensor(list(map(int, bitstring)))
+        if output_type == "torch"
+        else np.array(list(bitstring)).astype(int)
+    )
+
+
+def tensor_to_bitstring(bitstring: Tensor | np.array, output_type: str = "torch") -> str:
+    """
+    A helper function to convert torch.Tensor or numpy.array to bit strings.
+
+    Args:
+        bitstring: A torch.Tensor or numpy.array format of a bit string.
```

Comment:
Here the output_type definition (same from the above function) is missing.

Reply:
I would suggest to avoid numpy types as much as you can. Sorry for not pointing that earlier.

Reply:
I left these as an option, in case someone wants to use the bistring/tensor conversion as a standalone in the future. These are used with torch tensors internally.

Reply:
OK then, I will remove these.